### PR TITLE
Signup post

### PR DESCRIPTION
**Pull request by Fatmasiam**
#21

### CODE REVIEW

File: src/controllers/signup.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  
+  // const userData = req.body;
+  // addUser(userData)
+  // .then(result =>{
+  // res.send(JSON.stringify({ success : 'Regestration success ...'}));
+  // })
+  // .catch( err =>{
+  //   res.send(JSON.stringify({ error : 'There is error in networke '}));
+  // })
+}
```
Comment (Angham116) [inline src/controllers/signup.js]: why you pull the comments? 
and add a new line at the last of the file.

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const test = require('tape');
+const supertest = require('supertest');
+const router = require('../src/controllers');
+
+test('test signup route ', (t) => {
+  supertest(router)
```
Comment (Angham116) [inline test/test.js:44]: are you sure that the test is run right???

I think you should make supertest for app.js file, not router.js 

```javascript
const router = require('../src/app');
supertest(router)
```

**Review by Angham116 — CHANGES_REQUESTED** [review]

**Review by Angham116 — COMMENTED** [review]
On test.js line 12 : 
I think that
```t.error(err)``` not ```t.error = err```

File: src/controllers/signup.js (hunk 1 above)
Comment (Fatmasiam) [reply src/controllers/signup.js]: @Angham116  look to label  ? 
I think it is in progress

**Comment by Fatmasiam** [comment]
> On test.js line 12 :
> I think that
> `t.error(err)` not `t.error = err`

I now that and i have edit it put did not push it

File: src/controllers/signup.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+exports.post = (req, res) => {
+  const user = req.body;
+  const UserInfo = {
+    ...user.firstSection,
+    ...user.secondSection,
+    ...user.thirdSection,
+  };
+  console.log(UserInfo.password);
```
Comment (tomduggan85) [inline src/controllers/signup.js]: @Fatmasiam don't forget to remove this 😅

File: src/controllers/signup.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    error,
+  } = Joi.validate(UserInfo, signUpSchema);
+  if (!error) {
+    hashPassword(UserInfo.password).then((hashedPass) => {
+      UserInfo.password = hashedPass;
+      addUser(UserInfo)
+        .then(() => {
+          res.status(201).send(JSON.stringify({
+            success: 'Regestration success ...',
+            user: UserInfo,
```
Comment (tomduggan85) [inline src/controllers/signup.js]: @Fatmasiam For sake of security, I think you should remove `password` from the userInfo that you send back to the user

File: src/controllers/signup.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const UserInfo = {
+    ...user.firstSection,
+    ...user.secondSection,
+    ...user.thirdSection,
+  };
+  console.log(UserInfo.password);
+  const {
+    error,
+  } = Joi.validate(UserInfo, signUpSchema);
+  if (!error) {
```
Comment (tomduggan85) [inline src/controllers/signup.js:27]: @Fatmasiam You could add an `else` block to this if-statement and return a 400 (with the validate's error)

File: src/controllers/signup.js (hunk 3 above)
Comment (Fatmasiam) [reply src/controllers/signup.js]: @tomduggan85  I will remove it

File: src/controllers/signup.js (hunk 4 above)
Comment (Fatmasiam) [reply src/controllers/signup.js]: @tomduggan85  
thanx for your warning me on this point, 
I forget it.

**Pull request merged**